service-api/module/Application/src/Controller/YotiController.php

```diff
@@ -63,54 +64,76 @@ public function findPostOfficeAction(): JsonModel
-            ));
-        }
+        $this->getResponse()->setStatusCode(Response::STATUS_CODE_200);
+        $data = ['status' => $session['state']];
 
-        $this->getResponse()->setStatusCode(Response::STATUS_CODE_201);
-        return new JsonModel($result);
+        return new JsonModel($data);
     }
-
-    public function getSessionStatusAction(): JsonModel
+    /**
+     * @throws YotiException
+     */
+    public function initiateCounterServiceAction(): JsonModel
     {
         $uuid = $this->params()->fromRoute('uuid');
-
         if (! $uuid) {
             $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
             return new JsonModel(['error' => 'Missing uuid']);
         }
-        //@TODO look up actual sessionId from case and case where this is not created
-        $sessionId = 'AJDAHDFSH';
-        $session = $this->yotiService->retrieveResults($sessionId);
+        $notifyAuthToken = strval(Uuid::uuid4());
+        $caseData = $this->dataQuery->getCaseByUUID($uuid);
 
-        $this->getResponse()->setStatusCode(Response::STATUS_CODE_200);
-        $data = ['status' => $session['state']];
+        if (! $caseData) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
+            return new JsonModel(['error' => 'Case data not found']);
+        }
+        $sessionData = $this->sessionConfig->build($caseData, $notifyAuthToken);
+        $nonce = strval(Uuid::uuid4());
+        $dateTime = new DateTime();
+        $timestamp = $dateTime->getTimestamp();
 
-        return new JsonModel($data);
-    }
+        try {
+            $result = $this->yotiService->createSession($sessionData, $nonce, $timestamp);
+            $yotiSessionId = $result["data"]["session_id"];
 
-    public function getPDFLetterAction(string $session): JsonModel
-    {
-        $data = [];
-        $data['response'] = $this->yotiService->retrieveLetterPDF($session);
-        return new JsonModel($data);
+            if ($result["status"] < 400) {
+                $this->dataImportHandler->updateCaseData(
+                    $uuid,
+                    'sessionId',
+                    'S',
+                    $yotiSessionId
+                );
+            }
+            //save authToken to case
+            $this->dataImportHandler->updateCaseData(
+                $uuid,
+                'notifyAuthToken',
+                'S',
+                $notifyAuthToken
+            );
+            //Prepare and generate PDF
+            $this->yotiService->preparePDFLetter($caseData, $nonce, $timestamp, $yotiSessionId);
+            $this->yotiService->retrieveLetterPDF($yotiSessionId, $nonce, $timestamp);
+            //@TODO send pdf from above to sirius when ready
+        } catch (YotiException $e) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_500);
+            return new JsonModel(new Problem(
+                'Problem requesting Yoti API',
```

Comment:
Best to be specific about the problem, so "Could not generate and retrieve letter from Yoti API"

Reply:
This does then go onto show more specific messages from the 'errors' part from the exceptions that get bubbled up, so it could broadly be auth issue/signing or specific stages show after the 'Problem requesting Yoti API' message